spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerTable.java

```diff
@@ -44,38 +41,35 @@
  * SpannerTable implements Table.
  */
 public class SpannerTable implements Table, SupportsRead, SupportsWrite {
-  private String tableName;
+
+  private final String tableName;
 
   private final String instanceId;
 
   private final String databaseId;
 
   private final String projectId;
-  private StructType tableSchema;
+  private final SpannerTableSchema tableSchema;
   private static final ImmutableSet<TableCapability> tableCapabilities =
       ImmutableSet.of(TableCapability.BATCH_READ);
-  private static final String QUERY_PREFIX =
-      "SELECT COLUMN_NAME, ORDINAL_POSITION, IS_NULLABLE='YES' AS ISNULLABLE, SPANNER_TYPE "
-          + "FROM INFORMATION_SCHEMA.COLUMNS WHERE ";
-  private static final String QUERY_SUFFIX = " ORDER BY ORDINAL_POSITION";
-  private static final String GOOGLESQL_SCHEMA =
-      QUERY_PREFIX + "TABLE_NAME=@tableName" + QUERY_SUFFIX;
-  private static final String POSTGRESQL_SCHEMA =
-      QUERY_PREFIX + "columns.table_name=$1" + QUERY_SUFFIX;
 
   private static final Logger log = LoggerFactory.getLogger(SpannerTable.class);
 
   public SpannerTable(Map<String, String> properties) {
+    this.tableName =
```

Comment:
You may want to have a private method to `private String getOption(properties, option) { Objects.requireNonNull(properties.get(option), "Option \"<option>\" is missing."); }

Reply:
Thanks, that's a great idea.